jena-core/src/main/java/org/apache/jena/ext/xerces/impl/validation/ValidationState.java

```diff
@@ -169,16 +182,21 @@ public boolean isEntityUnparsed (String name) {
     // id
     @Override
     public boolean isIdDeclared(String name) {
+        if (fIdTable == null) {
+            return false;
+        }
         return fIdTable.containsKey(name);
     }
     @Override
     public void addId(String name) {
+        ensureIdTablesAreInitialized();
```

Comment:
Only init fIdTable on demand here. 
idRefTable is used in oder methods, so it might be that only one of the maps is needed.
```java
if(fIdTable == null) {
    fIdTable = new HashMap();
}
```